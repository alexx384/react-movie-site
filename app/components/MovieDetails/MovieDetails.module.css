.block {
  display: flex;
  margin: 0;
  text-align: left;
}

.poster {
  margin: 0px 60px 0px 0px;
  max-width: 25vw;
  object-fit: contain;
}

.title-and-rating {
  display: flex;
}

.title {
  margin: 0px 25px 0px 0px;
  color: hsl(0, 0%, 100%);
}

.rating {
  margin: 0;
  border: 1px solid hsl(0, 0%, 100%);
  border-radius: 100%;
  padding: 1.1rem 0.9rem 1.1rem 0.9rem;
  color: hsl(0, 0%, 100%);
}

.genre {
  margin: 4px 0px 0px 0px;
  color: hsla(0, 0%, 100%, 0.5);
}

.year-and-timing {
  display: flex;
  margin: 30px 0px 0px 0px;
}

.year {
  margin: 0px 51px 0px 0px;
  color: #f65261;
}

.timing {
  margin: 0;
  color: #f65261;
}

.description {
  margin: 29px 0px 0px 0px;
  color: hsla(0, 0%, 100%, 0.5);
}
