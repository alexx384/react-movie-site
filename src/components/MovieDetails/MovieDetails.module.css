@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@300;400;500&display=swap');

.movieDetails {
  margin: 0;
  display: flex;
  font-family: 'Montserrat', sans-serif;
}

.moviePoster {
  margin: 0px 60px 0px 0px;
}

.movieTitleAndRating {
  display: flex;
}

.movieTitle {
  font-family: inherit;
  font-size: 2.5rem;
  font-weight: 300;
  line-height: 1.5em;
  letter-spacing: 1px;
  text-align: left;
  color: #ffffff;
  text-transform: uppercase;
  margin: 0px 25px 0px 0px;
}

.movieRating {
  font-family: inherit;
  font-size: 1.25rem;
  font-weight: 300;
  line-height: 1.2em;
  letter-spacing: 0px;
  color: #ffffff;
  padding: 1.1rem 0.9rem 1.1rem 0.9rem;
  margin: 0;
  border: 1px solid #ffffff;
  border-radius: 100%;
}

.movieGenres {
  font-family: inherit;
  font-size: 0.9rem;
  font-weight: 500;
  line-height: 1.81em;
  letter-spacing: 0px;
  text-align: left;
  color: hsla(0, 0%, 100%, 0.5);
  margin: 4px 0px 0px 0px;
}

.movieYearAndTiming {
  margin: 30px 0px 0px 0px;
  display: flex;
  color: #f65261;
  font-family: Montserrat;
  font-size: 1.5rem;
  font-weight: 300;
  line-height: 1.208em;
  letter-spacing: 0px;
}

.movieYear {
  margin: 0px 51px 0px 0px;
  font-size: inherit;
  font-weight: inherit;
  line-height: inherit;
  letter-spacing: inherit;
}

.movieTiming {
  margin: 0;
  font-size: inherit;
  font-weight: inherit;
  line-height: inherit;
  letter-spacing: inherit;
}

.movieDescription {
  margin: 29px 0px 0px 0px;
  color: hsla(0, 0%, 100%, 0.5);
  font-family: inherit;
  font-size: 1.25rem;
  font-weight: 300;
  line-height: 1.2em;
  letter-spacing: 0px;
  text-align: left;
}
