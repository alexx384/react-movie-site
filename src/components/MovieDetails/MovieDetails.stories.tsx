import { ComponentStory, ComponentMeta } from '@storybook/react';
import { MovieDetails } from '.';
import imageURL from '../../assets/1.png';

export default {
  title: 'Stories/MovieDetails',
  component: MovieDetails,
} as ComponentMeta<typeof MovieDetails>;

const Template: ComponentStory<typeof MovieDetails> = (args) => (
  <MovieDetails {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  movieURL: imageURL,
  title: 'Pulp Fiction',
  releaseDate: new Date(1994, 0),
  rating: 8.9,
  genreIds: new Set(['Adventure']),
  runtime: 9240,
  overview:
    'Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra',
};
