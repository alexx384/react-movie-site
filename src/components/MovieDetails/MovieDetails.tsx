import React from 'react';
import {
  stringListToString,
  secondsToHoursAndMinutesString,
} from '../utils/string';
import styles from './MovieDetails.module.css';

type Props = {
  imageUrl: string;
  movieName: string;
  releaseYear: number;
  rating: number;
  genres: string[];
  durationInSeconds: number;
  description: string;
};

export const MovieDetails = ({
  imageUrl,
  movieName,
  releaseYear,
  rating,
  genres,
  durationInSeconds,
  description,
}: Props) => {
  return (
    <div className={styles.movieDetails}>
      <img className={styles.moviePoster} src={imageUrl} alt={movieName} />
      <div>
        <div className={styles.movieTitleAndRating}>
          <h1 className={styles.movieTitle}>{movieName}</h1>
          <h3 className={styles.movieRating}>{rating}</h3>
        </div>
        <h4 className={styles.movieGenres}>{stringListToString(genres)}</h4>
        <div className={styles.movieYearAndTiming}>
          <h2 className={styles.movieYear}>{releaseYear}</h2>
          <h2 className={styles.movieTiming}>
            {secondsToHoursAndMinutesString(durationInSeconds)}
          </h2>
        </div>
        <p className={styles.movieDescription}>{description}</p>
      </div>
    </div>
  );
};
