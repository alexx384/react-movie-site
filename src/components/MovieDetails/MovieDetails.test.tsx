import { MovieDetails } from './MovieDetails';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MovieDetails
        id="1"
        imageUrl="logo192.png"
        movieName="React"
        releaseYear={2022}
        rating={8.9}
        genre="Adventure"
        durationInSeconds={9240}
        description="Whether you work on your own or with thousands of other developers, using React feels the same. It is designed to let you seamlessly combine components written by independent people, teams, and organizations."
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
