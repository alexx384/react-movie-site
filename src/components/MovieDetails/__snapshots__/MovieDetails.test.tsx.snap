// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="movieDetails"
>
  <img
    alt="React"
    className="moviePoster"
    src="logo192.png"
  />
  <div>
    <div
      className="movieTitleAndRating"
    >
      <h1
        className="movieTitle"
      >
        React
      </h1>
      <h3
        className="movieRating"
      >
        8.9
      </h3>
    </div>
    <h4
      className="movieGenres"
    >
      Adventure
    </h4>
    <div
      className="movieYearAndTiming"
    >
      <h2
        className="movieYear"
      >
        2022
      </h2>
      <h2
        className="movieTiming"
      >
        2h 34min
      </h2>
    </div>
    <p
      className="movieDescription"
    >
      Whether you work on your own or with thousands of other developers, using React feels the same. It is designed to let you seamlessly combine components written by independent people, teams, and organizations.
    </p>
  </div>
</div>
`;
